test/Microsoft.AspNetCore.Html.Abstractions.Test/HtmlFormattableStringTest.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.WebEncoders.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Html
+{
+    public class HtmlFormattableStringTest
+    {
+        [Fact]
+        public void HtmlFormattableString_EmptyArgs()
+        {
+            // Arrange
+            var formattableString = new HtmlFormattableString("Hello, World!");
+
+            // Act
+            var result = HtmlContentToString(formattableString);
+
+            // Assert
+            Assert.Equal("Hello, World!", result);
+        }
+
+        [Fact]
+        public void HtmlFormattableString_EmptyArgsAndCulture()
+        {
+            // Arrange
+            var formattableString = new HtmlFormattableString(CultureInfo.CurrentCulture, "Hello, World!");
+
+            // Act
+            var result = HtmlContentToString(formattableString);
+
+            // Assert
+            Assert.Equal("Hello, World!", result);
+        }
+
+        [Fact]
+        public void HtmlFormattableString_MultipleArguments()
+        {
+            // Arrange
+            var formattableString = new HtmlFormattableString("{0} {1} {2} {3}!", "First", "Second", "Third", "Fourth");
+
+            // Act
+            var result = HtmlContentToString(formattableString);
+
+            // Assert
+            Assert.Equal(
+                "HtmlEncode[[First]] HtmlEncode[[Second]] HtmlEncode[[Third]] HtmlEncode[[Fourth]]!",
+                result);
+        }
+
+        [Fact]
+        public void HtmlFormattableString_WithHtmlEncodedString()
+        {
+            // Arrange
+            var formattableString = new HtmlFormattableString("{0}!", new HtmlEncodedString("First"));
```

Comment:
This is the only test I see involving an `IHtmlContent` implementation.
- Since the code special-cases `HtmlEncodedString`, should test another implementation as well.
- I don't see any tests mixing formats with an `IHtmlContent` implementation. What happens with `new HtmlFormattableString("{0,-20}!", new HtmlEncodedString("First"))` for example?